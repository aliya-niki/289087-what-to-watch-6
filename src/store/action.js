import {getFilmsByGenre} from '../utils';

export const ActionType = {
  CHANGE_ACTIVE_GENRE: `films/changeActiveGenre`,
  FILTER_FILMS_BY_GENRE: `films/filterFilmsByGenre`,
  RESET_SHOWN_FILMS_NUMBER: `films/resetShownFilmsNumber`,
  INCREASE_SHOWN_FILMS_NUMBER: `films/increaseShownFilmsNumber`,
  REQUIRE_AUTHORIZATION: `user/requireAuthorization`,
  LOAD_FILMS: `films/loadFilms`,
  LOAD_PROMO: `films/loadPromo`
};

export const ActionCreator = {
  changeActiveGenre: (genre) => ({
    type: ActionType.CHANGE_ACTIVE_GENRE,
    payload: genre,
  }),
  filterFilmsByGenre: (films, genre) => {
    const filteredFilms = getFilmsByGenre(films, genre);

    return {
      type: ActionType.FILTER_FILMS_BY_GENRE,
      payload: filteredFilms
    };
  },
  resetShownFilmsNumber: () => ({
    type: ActionType.RESET_SHOWN_FILMS_NUMBER,
  }),
  increaseShownFilmsNumber: () => ({
    type: ActionType.INCREASE_SHOWN_FILMS_NUMBER,
  }),
  requireAuthorization: (status) => ({
    type: ActionType.REQUIRE_AUTHORIZATION,
    payload: status,
  }),
  loadFilms: (films) => ({
    type: ActionType.LOAD_FILMS,
    payload: films,
  }),
  loadPromo: (promo) => ({
    type: ActionType.LOAD_PROMO,
    payload: promo,
  })
};
