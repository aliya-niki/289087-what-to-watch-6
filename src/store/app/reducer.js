import {ActionType} from './actions';
import {DEFAULT_ACTIVE_GENRE} from '../../const';

const initialState = {
  activeGenre: DEFAULT_ACTIVE_GENRE,
};

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_ACTIVE_GENRE:
      return {
        ...state,
        activeGenre: action.payload
      };
  }
  return state;
};
