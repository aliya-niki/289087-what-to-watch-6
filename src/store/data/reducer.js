import {ActionType} from './actions';
import {ReviewPostStatus} from '../../const';

const initialState = {
  films: [],
  promo: null,
  favorites: [],
  reviewPostStatus: ReviewPostStatus.PENDING,
};

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.LOAD_FILMS:
      return {
        ...state,
        films: action.payload,
      };
    case ActionType.LOAD_PROMO:
      return {
        ...state,
        promo: action.payload,
      };
    case ActionType.LOAD_FAVORITES:
      return {
        ...state,
        favorites: action.payload,
      };
    case ActionType.SET_REVIEW_POST_STATUS:
      return {
        ...state,
        reviewPostStatus: action.payload,
      };
  }
  return state;
};

