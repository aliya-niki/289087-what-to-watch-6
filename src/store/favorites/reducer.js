import {ActionType} from './actions';

export const initialState = {
  favorites: [],
};

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.LOAD_FAVORITES:
      return {
        ...state,
        favorites: action.payload,
      };
  }
  return state;
};

