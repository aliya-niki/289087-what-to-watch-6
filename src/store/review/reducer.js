import {ActionType} from './actions';
import {ReviewPostStatus} from '../../const';

export const initialState = {
  reviewPostStatus: ReviewPostStatus.PENDING,
};

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.SET_REVIEW_POST_STATUS:
      return {
        ...state,
        reviewPostStatus: action.payload,
      };
  }
  return state;
};

