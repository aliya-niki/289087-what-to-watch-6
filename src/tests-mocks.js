export const filmsRaw = [{
  "id": 1,
  "name": `The Grand Budapest Hotel`,
  "preview_image": `img/the-grand-budapest-hotel-poster.jpg`,
  "poster_image": `img/the-grand-budapest-hotel-poster.jpg`,
  "background_image": `img/bg-the-grand-budapest-hotel.jpg`,
  "background_color": `#ffffff`,
  "video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  "preview_video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  "description": `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  "rating": 8.9,
  "scores_count": 240,
  "director": `Wes Andreson`,
  "starring": [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`, `Tilda Swinton`, `Tom Wilkinson`, `Owen Wilkinson`, `Adrien Brody`, `Ralph Fiennes`],
  "run_time": 99,
  "genre": `Comedy`,
  "released": 2014,
  "is_favorite": true
}, {
  "id": 2,
  "name": `Bohemian Rhapsody`,
  "preview_image": `img/bohemian-rhapsody.jpg`,
  "poster_image": `img/the-grand-budapest-hotel-poster.jpg`,
  "background_image": `img/bg-the-grand-budapest-hotel.jpg`,
  "background_color": `#ffffff`,
  "video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  "preview_video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  "description": `Freddie Mercury -- the lead singer of Queen -- defies stereotypes and convention to become one of history's most beloved entertainers.`,
  "rating": 8.5,
  "scores_count": 240,
  "director": `Bryan Singer`,
  "starring": [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
  "run_time": 134,
  "genre": [`Drama`, `Biography`],
  "released": 2018,
  "is_favorite": true
}, {
  "id": 3,
  "name": `Macbeth`,
  "preview_image": `img/macbeth.jpg`,
  "poster_image": `img/the-grand-budapest-hotel-poster.jpg`,
  "background_image": `img/bg-the-grand-budapest-hotel.jpg`,
  "background_color": `#ffffff`,
  "video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  "preview_video_link": `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  "description": `After a brilliant victory in battle, Macbeth returns to an indolent court where honor is dispensed by whim.`,
  "rating": 7.9,
  "scores_count": 240,
  "director": `Kit Monkman`,
  "starring": [`Mark Rowley`, `Al Weaver`, `David Bradley`],
  "run_time": 121,
  "genre": `Drama`,
  "released": 2018,
  "is_favorite": false
}];

export const filmsAdapted = [{
  id: 1,
  name: `The Grand Budapest Hotel`,
  previewImage: `img/the-grand-budapest-hotel-poster.jpg`,
  posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#ffffff`,
  videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  rating: 8.9,
  scoresCount: 240,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`, `Tilda Swinton`, `Tom Wilkinson`, `Owen Wilkinson`, `Adrien Brody`, `Ralph Fiennes`],
  runTime: 99,
  genre: `Comedy`,
  released: 2014,
  isFavorite: true
}, {
  id: 2,
  name: `Bohemian Rhapsody`,
  previewImage: `img/bohemian-rhapsody.jpg`,
  posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#ffffff`,
  videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  description: `Freddie Mercury -- the lead singer of Queen -- defies stereotypes and convention to become one of history's most beloved entertainers.`,
  rating: 8.5,
  scoresCount: 240,
  director: `Bryan Singer`,
  starring: [`Rami Malek`, `Lucy Boynton`, `Gwilym Lee`],
  runTime: 134,
  genre: [`Drama`, `Biography`],
  released: 2018,
  isFavorite: true
}, {
  id: 3,
  name: `Macbeth`,
  previewImage: `img/macbeth.jpg`,
  posterImage: `img/the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#ffffff`,
  videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  description: `After a brilliant victory in battle, Macbeth returns to an indolent court where honor is dispensed by whim.`,
  rating: 7.9,
  scoresCount: 240,
  director: `Kit Monkman`,
  starring: [`Mark Rowley`, `Al Weaver`, `David Bradley`],
  runTime: 121,
  genre: `Drama`,
  released: 2018,
  isFavorite: false
}];

export const authRaw = {
  "id": 1,
  "email": `keks@mail.com`,
  "name": `keks`,
  "avatar_url": `keks.jpeg`,
};

export const authAdapted = {
  id: 1,
  email: `keks@mail.com`,
  name: `keks`,
  avatarUrl: `keks.jpeg`,
};

export const user = {
  email: `keks@mail.com`,
  password: 12345678
};

export const reviews = [
  {
    id: 1,
    user: {
      id: 111,
      name: `Kate Fogler`
    },
    rating: 9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  }, {
    id: 2,
    user: {
      id: 287,
      name: `Kate Muir`
    },
    rating: 6,
    comment: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
    date: `2019-05-08T14:13:56.569Z`
  }, {
    id: 3,
    user: {
      id: 17,
      name: `Kate Hidi`
    },
    rating: 2,
    comment: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
    date: `2019-05-08T14:13:56.569Z`
  }, {
    id: 4,
    user: {
      id: 43,
      name: `Kate Snow`
    },
    rating: 4,
    comment: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    date: `2019-05-08T14:13:56.569Z`
  }
];
