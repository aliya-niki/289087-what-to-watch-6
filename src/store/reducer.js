import {ActionType} from './action';
import {DEFAULT_ACTIVE_GENRE, MOVIES_NUMBER_PER_STEP, AuthorizationStatus} from '../const';
import reviews from '../mocks/reviews';
import {getFilmsByGenre} from '../utils';

const initialState = {
  activeGenre: DEFAULT_ACTIVE_GENRE,
  films: [],
  reviews,
  promo: {
    id: 0,
    name: ``,
    previewImage: ``,
    posterImage: ``,
    backgroundImage: ``,
    backgroundColor: ``,
    videoLink: ``,
    previewVideoLink: ``,
    description: ``,
    rating: 0,
    scoresCount: 0,
    director: ``,
    starring: [],
    runTime: 0,
    genre: ``,
    released: 0,
    isFavorite: false
  },
  filteredFilms: [],
  shownFilmsNumber: MOVIES_NUMBER_PER_STEP,
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  isFilmsDataLoaded: false
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_ACTIVE_GENRE:
      return {
        ...state,
        activeGenre: action.payload
      };
    case ActionType.FILTER_FILMS_BY_GENRE:
      return {
        ...state,
        filteredFilms: action.payload
      };
    case ActionType.RESET_SHOWN_FILMS_NUMBER:
      return {
        ...state,
        shownFilmsNumber: MOVIES_NUMBER_PER_STEP
      };
    case ActionType.INCREASE_SHOWN_FILMS_NUMBER:
      return {
        ...state,
        shownFilmsNumber: state.shownFilmsNumber + MOVIES_NUMBER_PER_STEP
      };
    case ActionType.REQUIRE_AUTHORIZATION:
      return {
        ...state,
        authorizationStatus: action.payload,
      };
    case ActionType.LOAD_FILMS:
      return {
        ...state,
        films: action.payload,
        filteredFilms: getFilmsByGenre(action.payload, state.activeGenre),
        isFilmsDataLoaded: true,
      };
    case ActionType.LOAD_PROMO:
      return {
        ...state,
        promo: action.payload,
      };
  }

  return state;
};

export {reducer};
