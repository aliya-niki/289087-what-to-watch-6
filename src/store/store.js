import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import {composeWithDevTools} from 'redux-devtools-extension';
import {createAPI} from '../services/api';
import reducer from './reducer';
import {setUserInfo} from './user/actions';
import {redirect} from './middlewares/redirect';

const api = createAPI(
    () => store.dispatch(setUserInfo(null))
);

const store = createStore(
    reducer,
    composeWithDevTools(
        applyMiddleware(thunk.withExtraArgument(api)),
        applyMiddleware(redirect)
    )
);

export default store;
