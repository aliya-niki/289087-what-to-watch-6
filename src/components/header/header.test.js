import React from 'react';
import {render} from '@testing-library/react';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import configureStore from 'redux-mock-store';
import * as redux from 'react-redux';
import {NameSpace} from '../../store/reducer';
import Header from './header';
import {authAdapted} from '../../tests-mocks';

const mockStore = configureStore({});

describe(`Header`, () => {
  it(`should render correctly with authorized user`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationError: false,
        userAuthorizationInfo: authAdapted,
      },
    });
    const history = createMemoryHistory();
    const {container} = render(
        <redux.Provider store={store}>
          <Router history={history}>
            <Header additionalClassName={`movie-card__head`}/>
          </Router>
        </redux.Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`should render correctly with unauthorized user`, () => {
    const store = mockStore({
      [NameSpace.USER]: {
        authorizationError: false,
        userAuthorizationInfo: null,
      },
    });
    const history = createMemoryHistory();
    const {container} = render(
        <redux.Provider store={store}>
          <Router history={history}>
            <Header additionalClassName={`movie-card__head`}/>
          </Router>
        </redux.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
