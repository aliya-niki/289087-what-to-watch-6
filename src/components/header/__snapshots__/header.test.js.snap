// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly with authorized user 1`] = `
<div>
  <header
    class="page-header movie-card__head"
  >
    <div
      class="logo"
    >
      <a
        class="logo__link"
        href="/"
      >
        <span
          class="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          class="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          class="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      class="user-block"
    >
      <a
        href="/mylist"
      >
        <div
          class="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="keks.jpeg"
            width="63"
          />
        </div>
      </a>
    </div>
  </header>
</div>
`;

exports[`Header should render correctly with unauthorized user 1`] = `
<div>
  <header
    class="page-header movie-card__head"
  >
    <div
      class="logo"
    >
      <a
        class="logo__link"
        href="/"
      >
        <span
          class="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          class="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          class="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      class="user-block"
    >
      <a
        class="user-block__link"
        href="/login"
      >
        Sign in
      </a>
    </div>
  </header>
</div>
`;
